app/postgres/users.go

```diff
@@ -0,0 +1,35 @@
+package postgres
+
+import (
+	"database/sql"
+
+	"github.com/OAuth2withJWT/identity-provider/app"
+)
+
+type UserRepository struct {
+	db *sql.DB
+}
+
+func NewUserRepository(db *sql.DB) *UserRepository {
+	return &UserRepository{
+		db: db,
+	}
+}
+
+func (ur *UserRepository) Create(req app.CreateUserRequest) (*app.User, error) {
+	hashedPassword, err := app.HashPassword(req.Password)
```

Comment:
I would just move this password hashing part back to the `UserService.Create` function. The repository functions only have the responsibility of performing the basic operation with the repository (in this case creating the user in the postgresql database) and return the result. Imagine if you added another implementation of the repository (let's say a file repository or whatever) you would have to reimplement the hashing logic. Because you always want the password to be hashed, you would put it in the service function.